vm/src/org/mozartoz/truffle/nodes/call/SelfTailCallCatcherNode.java

```diff
@@ -0,0 +1,81 @@
+package org.mozartoz.truffle.nodes.call;
+
+import org.mozartoz.truffle.nodes.OzNode;
+import org.mozartoz.truffle.runtime.OzBacktrace;
+import org.mozartoz.truffle.runtime.SelfTailCallException;
+
+import com.oracle.truffle.api.Truffle;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.LoopNode;
+import com.oracle.truffle.api.nodes.RepeatingNode;
+
+public class SelfTailCallCatcherNode extends OzNode {
+
+	OzNode body;
+	@Child LoopNode loopNode;
+
+	public SelfTailCallCatcherNode(OzNode body) {
+		this.body = body;
```

Comment:
The Serializer just does not seem to work wihout it, is there a way to get rid of it?